### PR TITLE
Enable Bulk Delete

### PR DESCRIPTION
**Pull request by tonysun83**
This is a port of COUCHDB--2651.
Add _bulk_delete path to _index so that users can bulk delete
indexes via POST. Users will pass in a list of docids via the POST body.
We delete each index one by one until an error is thrown or all indexes
in the list are deleted. We return a list of successful deletes
and failed deletes along with the ids. For a single delete, we still
throw the same errors as before to be backwards compatible
with any applications expecting an error.

FogBugID:46243

### CODE REVIEW

File: src/mango_error.erl [hunk 1]
```diff
@@ -110,6 +110,7 @@ info(mango_idx, {invalid_index_type, BadType}) ->
         <<"invalid_index">>,
         fmt("Invalid type for index: ~s", [BadType])
     };
+
```
Comment (davisp) [inline src/mango_error.erl]: Stray whitespace changes should be in their own commit.

Comment (davisp) [inline src/mango_error.erl]: s/stray whitespace/whitespace only changes/

**Comment by davisp** [comment]
This seems like an awfully big change in order to reuse some of the code that exists for removing indexes. Given that you're removing the entire design document it seems like we might be able to reduce a lot of the complexity here by just adding a mango_idx:bulk_delete/3 call that would do the deletion using a bulk fabric:update_docs/3 call.

**Comment by tonysun83** [comment]
How descriptive is the response from fabric:update_docs/3 call? Diving into the code, it does seem that it's call to couch_db:update_docs/4 will eventually return a list of successful and failed commits. If so, I would just have to massage that message for the user? I just want to make sure the user gets a proper detailed list rather than just a fail or success.

**Comment by davisp** [comment]
It'll give you a response per doc. Its the same code that drives _bulk_docs which has similar behavior.

**Comment by tonysun83** [comment]
Closing this one since we're going with a different approach than this PR.
